packages/flutter/lib/src/rendering/object.dart

```diff
@@ -819,6 +834,146 @@ class PipelineOwner {
+  /// O(n^2) but worse than O(n log(n)) (its curve is similar to O(n^1.8)). If
+  /// the check degrades performance unacceptably for debug usage, consider
+  /// setting [maxElevationObjectsToCheck] to a lower number, or to 0 to disable
+  /// it completely. The number can still be set higher in unit tests, where
+  /// performance is less critical.
+  Path debugCheckElevationData(
+    double elevation,
+    Rect area,
+    Path path,
+    RenderObject object,
+    RenderObject lastCommonAncestor,
+  ) {
+    assert(elevation != null);
+    assert(area != null);
+    assert(path != null);
+    assert(object != null);
+    _elevations[lastCommonAncestor] ??= <_ElevationData>[];
+    // Check in reverse order - we're more likely to fail on a closer node
+    // if we're going to fail at all. Take only _maxElevationObjectsToCheck
+    // to avoid this taking too long in the paint cycle.
+    final Iterable<_ElevationData> elevationsToCheck = _elevations[lastCommonAncestor]
+        .reversed
+        .take(_maxElevationObjectsToCheck)
+        .where((_ElevationData elevationData) {
+          return elevation < elevationData.elevation &&
+                 area.overlaps(elevationData.area);
+        });
+    for (final _ElevationData elevationData in elevationsToCheck) {
+      final Path difference = Path.combine(
+        PathOperation.union,
+        path,
+        _translatePath(elevationData.path, elevationData.area.topLeft - area.topLeft),
+      );
+      // Check if the union created a path with a single segment. If so, the
+      // paths overlap.
+      // This will not work if we're dealing with input paths that are already
+      // multi-segment shape, but checking for the segments of the incoming paths
+      // adds complexity and cost for an unusual case.
+      final ui.PathMetrics differenceMetrics = difference.computeMetrics();
+      // Don't use `PathMetrics.length`, we don't actually want to iterate the
+      // whole set. Instead, just make sure we've ended up with one distinct
+      // contour.
+      differenceMetrics.iterator.moveNext();
+      if (!differenceMetrics.iterator.moveNext()) {
+        object._debugReportException(
+          'paint',
+          'An attempt was made to paint a $object with an '
+          'elevation of $elevation after a ${elevationData.object} with an '
+          'elevation of ${elevationData.elevation} in the same area of the '
+          'screen.\n\n'
+          'This can happen when placing multiple children that have '
+          'elevations in a Stack or CustomMultiChildLayout widget and '
+          'painting them out of order with respect to their elevations.\n\n'
+          'This is not a valid use of elevation, and will cause rendering '
+          'inconsistencies on platforms that use the elevation property to '
+          'in ways that affect painting order.',
+          null, // The StackTrace is very unhelpful here.
```

Comment:
Does _debugReportException handle this being null graciously in a non-confusing way to the user?

Reply:
It just doesn't print anything for the stacktrace portion.  Otherwise the StackTrace.current for this ends up being 200+ lines of internal painting logic.